### ISSUE TITLE
Exception when writing Assembly (doesn't happen with dnlib)

### ISSUE THREAD

**Issue by vikigenius** [Washi1337/AsmResolver#208]
## Describe the bug
I was trying to publicize the members of the assembly as per my previous issue: https://github.com/Washi1337/AsmResolver/issues/206
That threw an exception

```
System.AggregateException: Construction of the PE image failed with one or more errors. ---> AsmResolver.DotNet.Code.Cil.InvalidCilInstructionException: [System.Void Kingmaker.Utility.ReportCombatLogManager::AsyncFileWrite(System.IO.StreamWriter) (0x06002264)]: Handler End of exception handler 1 references an offset that is not present in the method body.
   at AsmResolver.DotNet.Code.Cil.CilLabelVerifier.Verify()
   at AsmResolver.DotNet.Code.Cil.CilMethodBody.VerifyLabels(Boolean calculateOffsets)
   at AsmResolver.DotNet.Code.Cil.CilMethodBodySerializer.SerializeMethodBody(MethodBodySerializationContext context, MethodDefinition method)
   --- End of inner exception stack trace ---
   at AsmResolver.DotNet.ModuleDefinition.ToPEImage(IPEImageBuilder imageBuilder)
   at AsmResolver.DotNet.ModuleDefinition.Write(IBinaryStreamWriter writer, IPEImageBuilder imageBuilder, IPEFileBuilder fileBuilder)
   at AsmResolver.DotNet.ModuleDefinition.Write(Stream outputStream, IPEImageBuilder imageBuilder, IPEFileBuilder fileBuilder)
   at AsmResolver.DotNet.ModuleDefinition.Write(String filePath, IPEImageBuilder imageBuilder, IPEFileBuilder fileBuilder)
   at AsmResolver.DotNet.ModuleDefinition.Write(String filePath)
   at AssemblyPublicizerTest.PatcherTest.TestAssemblyPatch() in C:\Users\maste\Documents\Modding\PWOTR\Mods\Dev\AssemblyPublicizer\AssemblyPublicizerTest\PatcherTest.cs:line 17
---> (Inner Exception #0) AsmResolver.DotNet.Code.Cil.InvalidCilInstructionException: [System.Void Kingmaker.Utility.ReportCombatLogManager::AsyncFileWrite(System.IO.StreamWriter) (0x06002264)]: Handler End of exception handler 1 references an offset that is not present in the method body.
   at AsmResolver.DotNet.Code.Cil.CilLabelVerifier.Verify()
   at AsmResolver.DotNet.Code.Cil.CilMethodBody.VerifyLabels(Boolean calculateOffsets)
   at AsmResolver.DotNet.Code.Cil.CilMethodBodySerializer.SerializeMethodBody(MethodBodySerializationContext context, MethodDefinition method)<---

---> (Inner Exception #1) AsmResolver.DotNet.Code.Cil.InvalidCilInstructionException: [System.Void Kingmaker.Utility.ReportingUtils::Sender() (0x060022BC)]: Handler End of exception handler 1 references an offset that is not present in the method body.
   at AsmResolver.DotNet.Code.Cil.CilLabelVerifier.Verify()
   at AsmResolver.DotNet.Code.Cil.CilMethodBody.VerifyLabels(Boolean calculateOffsets)
   at AsmResolver.DotNet.Code.Cil.CilMethodBodySerializer.SerializeMethodBody(MethodBodySerializationContext context, MethodDefinition method)<---

```

I didn't face this issue in dnlib. Here is the dll in question.
[ REDACTED ]

## Platform
 - OS: Windows 10
 - AsmResolver Version: 4.6.0

**Comment by vikigenius** [comment]
Ok on further testing, it seems like simply loading and writing the module also causes an error.

```
var module = ModuleDefinition.FromFile(refPath);
// AssemblyPatcher.Patch(module, true);
module.Write(targetPath);
```

Is there anyway I can ignore these error ?

**Comment by Washi1337** [comment]
Be careful with uploading binaries that are part of games or other types of licensed third party software to GitHub. I do not wish to violate any licenses or redistribution terms associated to these types of files. Next time, please send me the binary through e-mail or make a sample binary on your own :).

As for the bug, seems like you triggered a few edges cases in AsmResolver with this specific binary. I have implemented a potential fix on branch [hotfix/4.7.1](https://github.com/Washi1337/AsmResolver/tree/hotfix/4.7.1). Could you verify that this indeed fixes your issue? You can download the packages from [AppVeyor](https://ci.appveyor.com/project/Washi1337/asmresolver/builds/41002353/artifacts).

As for ignoring reader and writer errors, please refer to the [wiki](https://asmresolver.readthedocs.io/en/latest/faq.html#why-does-asmresolver-throw-so-many-errors-on-reading-writing). Note that in this case however, it is a bug in AsmResolver, so disabling errors will still result in an invalid binary being produced.

**Comment by vikigenius** [comment]
Yep, seems to have worked. I would have produced my own binary. But I couldn't isolate the issue for proper repro. Next time I will send you dll by email on request. Thanks.

**Comment by Washi1337** [comment]
Perfect, then this will be merged relatively soon be included in the 4.7.1 hotfix, and pushed to nuget.
